File: truss-compiler/src/main/java/truss/compiler/symbols/GlobalSymbol.java
package truss.compiler.symbols;

public class GlobalSymbol extends ContainerSymbol {
    public GlobalSymbol() {
        super(null);
    }

    @Override
    public SymbolKind getKind() {
        return SymbolKind.GLOBAL;
    }

    @Override
    public void accept(SymbolVisitor visitor) {
        if (!visitor.isDone()) {
            visitor.visitGlobal(this);
        }
    }
}

File: Truss.Compiler/Symbols/GlobalSymbol.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Truss.Compiler.Symbols {
    public class GlobalSymbol : ContainerSymbol {
        public GlobalSymbol()
            : base(null) {
        }

        public override SymbolKind Kind {
            get { return SymbolKind.Global; }
        }

        public override void Accept(ISymbolVisitor visitor) {
            if (!visitor.Done) {
                visitor.VisitGlobal(this);
            }
        }
    }
}
